packages/graphql/src/translate/queryAST/factory/OperationFactory.ts

```diff
@@ -60,42 +62,33 @@ export class OperationsFactory {
         entityOrRel: ConcreteEntityAdapter | RelationshipAdapter,
         resolveTree: ResolveTree,
         context: Neo4jGraphQLTranslationContext
-    ): ReadOperation {
-        const entity = (
-            entityOrRel instanceof RelationshipAdapter ? entityOrRel.target : entityOrRel
-        ) as ConcreteEntityAdapter;
-        const projectionFields = { ...resolveTree.fieldsByTypeName[entity.name] };
+    ): ReadOperation | InterfaceReadOperation {
+        const entity = entityOrRel instanceof RelationshipAdapter ? entityOrRel.target : entityOrRel;
```

Comment:
We can use the type-guard function isConcreteEntity to check only once if entityOrRel is an instance of `RelationshipAdapter`